### PR TITLE
docs: specify what happens when there are rules with the same name in an extended config

### PR DESCRIPTION
**Pull request by Greesb**
Fixes MRGFY-3996

### CODE REVIEW

File: src/content/docs/configuration/sharing.mdx [hunk 1]
```diff
@@ -93,4 +93,9 @@ pull_request_rules:
 
   - Values in the `default` key will be merged and remote default values will
     apply to local configuration.
+
+  - If there are `pull_request_rules`, `queue_rules`, `priority_rules`, `partition_rules`
+    or `merge_protections` rules with the same name in both the extended config and the
+    config from which it is extended, then the rule, with the same name, in the config that
+    extends the other will NOT be overwritten.
```
Comment (Syffe) [inline src/content/docs/configuration/sharing.mdx:100]:
```suggestion
  - If there are `pull_request_rules`, `queue_rules`, `priority_rules`, `partition_rules`
    or `merge_protections` rules with the same name in both the extended config and the
    config from which it is extended, then the duplicated rule will NOT be overwritten. Thus, the rule
    defined in the config containing the `extends` keyword will be used.
```
I had some troubles understanding, may be we can turn it in another way ?

My suggestion is just an example we can change it to be clearer, wdyt ?

**Review by Syffe — APPROVED** [review]

File: src/content/docs/configuration/sharing.mdx [hunk 2]
```diff
@@ -93,4 +93,9 @@ pull_request_rules:
 
   - Values in the `default` key will be merged and remote default values will
     apply to local configuration.
+
+  - If there is rule in any of the `pull_request_rules`, `queue_rules`, `priority_rules`, `partition_rules`
+    or `merge_protections` section with the same name in both the extended config and the
```
Comment (lecrepont01) [inline src/content/docs/configuration/sharing.mdx:98]: what do you think of introducing the words `parent config` and `child config` to make the concept easier to grasp?

Comment (Greesb) [reply src/content/docs/configuration/sharing.mdx]: Good idea, made a modification, let me know if that sounds good know

File: src/content/docs/configuration/sharing.mdx [hunk 3]
```diff
@@ -93,4 +93,8 @@ pull_request_rules:
 
   - Values in the `default` key will be merged and remote default values will
     apply to local configuration.
+
+  - If there is rule in any of the `pull_request_rules`, `queue_rules`, `priority_rules`, `partition_rules`
+    or `merge_protections` section with the same name in both the parent and child config, then the duplicated
+    rule in the child config will NOT be overwritten with rule from the parent config.
```
Comment (sileht) [inline src/content/docs/configuration/sharing.mdx:99]: child/parent wording is confusing here, I have no idea if the child is the one from `extends` or not.

Comment (Greesb) [reply src/content/docs/configuration/sharing.mdx]: Modified it to describe which is which

**Review by lecrepont01 — APPROVED** [review]

File: src/content/docs/configuration/sharing.mdx [hunk 4]
```diff
@@ -93,4 +93,9 @@ pull_request_rules:
 
   - Values in the `default` key will be merged and remote default values will
     apply to local configuration.
+
+  - If there is rule in any of the `pull_request_rules`, `queue_rules`, `priority_rules`, `partition_rules`
```
Comment (Syffe) [inline src/content/docs/configuration/sharing.mdx:97]:
```suggestion
  - If there is a rule in any of the `pull_request_rules`, `queue_rules`, `priority_rules`, `partition_rules`
```

**Review by Syffe — APPROVED** [review]

File: src/content/docs/configuration/sharing.mdx [hunk 5]
```diff
@@ -93,4 +93,9 @@ pull_request_rules:
 
   - Values in the `default` key will be merged and remote default values will
     apply to local configuration.
+
+  - If there is a rule in any of the `pull_request_rules`, `queue_rules`, `priority_rules`, `partition_rules`
```
Comment (DouglasBlackwood) [inline src/content/docs/configuration/sharing.mdx:97]:
```suggestion
  - If there is a rule in any of the `pull_request_rules`, `queue_rules`, `priority_rules`, `partition_rules`,
```

**Review by DouglasBlackwood — COMMENTED** [review]
Not clear IMO. We should simplify the sentence. Something like "a rule cannot be overwritten by another rule from the `extends` configuration, even if they have the same name".

Avoid the word "config", use "configuration".

File: src/content/docs/configuration/sharing.mdx (hunk 5 above)
Comment (Greesb) [reply src/content/docs/configuration/sharing.mdx:97]: Since there's a `or` after this word, i don't think a comma is here appropriate @DouglasBlackwood

Comment (DouglasBlackwood) [reply src/content/docs/configuration/sharing.mdx]: True in French, not in English 😄

Comment (Greesb) [reply src/content/docs/configuration/sharing.mdx]: I was intrigued so i searched up a bit, and looks like both are valid syntax IIUC: https://www.grammarly.com/blog/comma-before-or/ @DouglasBlackwood

Comment (Greesb) [reply src/content/docs/configuration/sharing.mdx]: Added the comma

Comment (DouglasBlackwood) [reply src/content/docs/configuration/sharing.mdx]: Interesting, thanks for the link.

**Review by DouglasBlackwood — APPROVED** [review]

**Review by Syffe — APPROVED** [review]

**Pull request merged**